source/common/stream_info/stream_info_impl.h

```diff
@@ -349,6 +349,7 @@ struct StreamInfoImpl : public StreamInfo {
     downstream_transport_failure_reason_ = std::string(info.downstreamTransportFailureReason());
     bytes_retransmitted_ = info.bytesRetransmitted();
     packets_retransmitted_ = info.packetsRetransmitted();
+    should_drain_connection_ = info.shouldDrainConnectionUponCompletion();
```

Comment:
don't we have to carry this over iff the flag is true?

Reply:
good catch! Fixed